File: typescript/streaming.ts
import { VecTrade } from "@vectrade/sdk";

async function main() {
  const vt = new VecTrade();

  console.log("=== Streaming AI Analysis ===\n");
  console.log("Prompt: Analyze MSFT cloud business outlook\n");
  console.log("-".repeat(50));

  for await (const chunk of vt.ai.stream("Analyze MSFT cloud business outlook for 2025-2026")) {
    process.stdout.write(chunk.text);
  }

  console.log("\n" + "-".repeat(50));
  console.log("\nDone!");
}

main().catch(console.error);

File: python/streaming.py
"""VecTrade Python SDK — Streaming AI Analysis Example."""

from vectrade import VecTrade


def main():
    vt = VecTrade()

    print("=== Streaming AI Analysis ===\n")

    # Stream a detailed analysis
    print("Prompt: Analyze NVDA before Q4 earnings\n")
    print("-" * 50)

    for chunk in vt.ai.stream("Analyze NVDA before Q4 earnings. Cover: revenue growth, AI segment, valuation, risks."):
        print(chunk.text, end="", flush=True)

    print("\n" + "-" * 50)
    print("\nDone!")


if __name__ == "__main__":
    main()
